src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -146,16 +144,14 @@ public boolean canUseLocalCache() {
      */
     private boolean verifyManifestPermissions() {
         PackageManager pm = mContext.getPackageManager();
-        boolean permission = PackageManager.PERMISSION_GRANTED == pm.checkPermission(
-                "android.permission.GET_ACCOUNTS", mContext.getPackageName())
-                && PackageManager.PERMISSION_GRANTED == pm.checkPermission(
-                        "android.permission.MANAGE_ACCOUNTS", mContext.getPackageName())
-                && PackageManager.PERMISSION_GRANTED == pm.checkPermission(
-                        "android.permission.USE_CREDENTIALS", mContext.getPackageName());
+        boolean permission = PackageManager.PERMISSION_GRANTED == pm.checkPermission("android.permission.GET_ACCOUNTS",
```

Comment:
1) There is method ContextCompat.checkSelfPermission 
2) It seems like this permissions will be deprecated in Android N
3) Why do we need MANAGE_ACCOUNTS permission from app that isn't broker? Can it manage accounts?
4) Not sure why you moving first argument to line above, IMHO previous version showed number of arguments better.